lahja/endpoint.py

```diff
@@ -190,6 +205,20 @@ def start_serving_nowait(self,
             loop=self.event_loop
         )
 
+    def _compress_event(self, event: BaseEvent) -> Union[BaseEvent, bytes]:
+        if self.has_snappy_support:
```

Comment:
Wondering if we could gain a little more by removing the negotiation and do all of this in the `__init__`.  Rough pseudo-code sketch:

```python
def __init__(self, ...).
    if is_snappy_available(...):
        self.compress = cytoolz.compose(snappy.compress, pickle.dumps)
        self.decompress = cytoolz.compose(pickle.loads, snappy.decompress)
    else:
        self.compress = cytoolz.identity
        self.decompress = cytoolz.identity
```

Reply:
I tried it but I couldn't measure any notable difference in speed.